### PR TITLE
Upgrade to golangci-lint from gometalinter

### PR DESCRIPTION
**Pull request by daxmc99**
Problem: gometallinter has been deprecated

Solution: Switch to gometalinter and enable to least number of linters
to be comparable to gometalinter.
See https://github.com/rancher/rancher/issues/18638

### CODE REVIEW

**Comment by dramich** [comment]
Per our discussion this pr should just get the new tool running using our current list of linters. Once this is merged we can look at turning on additional linters and fixing issues they create.

File: .golangci.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			"deadcode",
+			"varcheck"
+			
+		]
+	},
+	"linters-settings": {
+		"govet":{
+			"check-shadowing":false
+		},
+		"errcheck":{
```
Comment (dramich) [inline .golangci.json]: Drop this, the linter isn't enabled

File: .golangci.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"run": {
+		"tests": false,
+        "deadline": "10m",
+		"max-issues-per-linter": 0
+		
+	},
+	"issues":{
+		"exclude-rules":[
+			{   
+				"linters":"ineffassign",
```
Comment (dramich) [inline .golangci.json]: Drop this, the linter isn't enabled

**Comment by cjellick** [comment]
LGTM

**Comment by daxmc99** [comment]
Reverted by c0e17880540a115151c5edb8712e06dea1906da4

**Pull request merged**